src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -34,26 +39,32 @@ public static class TagHelperDescriptorFactory
         /// <param name="assemblyName">The assembly name that contains <paramref name="type"/>.</param>
         /// <param name="type">The type to create a <see cref="TagHelperDescriptor"/> from.</param>
         /// <returns>A <see cref="TagHelperDescriptor"/> that describes the given <paramref name="type"/>.</returns>
-        public static IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName, Type type)
+        public static IEnumerable<TagHelperDescriptor> CreateDescriptors(
+            string assemblyName,
+            Type type,
+            ParserErrorSink errorSink)
         {
-            var tagNames = GetTagNames(type);
+            var elementTargets = GetElementTargets(type, errorSink);
```

Comment:
no longer clear why `GetTagNames()` / `GetElementTargets()` is separate from `CreateDescriptors()`.  specifically, if you pass `typeName` and `attributeDescriptors` into `GetElementTargets()`, the `Select()` below and the intermediate `ElementTarget` type would not be needed.